### PR TITLE
Install latest src version in executors

### LINKED ISSUE #46219: Install the latest compatible src-cli version
Today, when running the command `executor install src-cli`, the minimum compatible version is downloaded and installed. The minimum version is specified by a [constant](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@v4.2.1/-/blob/internal/src-cli/consts.go?L9). This can cause issues when a user runs `executor validate`.

During validation, the version of `src` is checked against the _latest_ compatible version (by checking GitHub releases). Meaning if minimum supported version is behind the latest patch version, the validation check will fail.

We have a [TODO](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/enterprise/cmd/executor/internal/run/install.go?L299) to install the _latest_ compatible version.

## Done

- Update the install command for `src` to use the _latest_ compatible version instead of the minimum supported version
- If the version of `src` that is installed meets the _minimum_ supported version, do not error out.
    - Log a warning telling the user that they should upgrade to the _latest_ compatible version

### PR DESCRIPTION
**Pull request by Piszmog**
Closes #46219.

Made the following changes,

- Go to Sourcegraph to attempt to get the latest `src` version. Use this version when installing `src`
    - If an error occurs when trying to fetch the latest version, fallback to `srccli.MinimumVersion`
- If there is a patch mismatch, log a warning instead of stopping the process

## Test plan

Added unit tests.

### CODE REVIEW

File: enterprise/cmd/executor/internal/run/run.go [hunk 1]
```diff
@@ -64,8 +65,13 @@ func StandaloneRunRun(ctx context.Context, logger log.Logger, cfg *config.Config
 		if err != nil {
 			return err
 		}
-		if err := validateSrcCLIVersion(ctx, client, opts.QueueOptions.BaseClientOptions.EndpointOptions); err != nil {
-			return err
+		if err = validateSrcCLIVersion(ctx, client, opts.QueueOptions.BaseClientOptions.EndpointOptions); err != nil {
+			if errors.Is(err, ErrSrcPatchBehind) {
+				// This is ok. The patch just doesn't match but still works.
+				logger.Warn("Src patch version mismatch", log.Error(err))
```
Comment (eseliger) [inline enterprise/cmd/executor/internal/run/run.go:71]: can we add some more context to what that means in the log msg?
```suggestion
				logger.Warn("A newer patch release version of src-cli is available, consider running executor install src-cli to upgrade", log.Error(err))
```

Comment (eseliger) [reply enterprise/cmd/executor/internal/run/run.go:71]: the same in validate

File: enterprise/cmd/executor/internal/run/run.go [hunk 2]
```diff
@@ -64,8 +65,13 @@ func StandaloneRunRun(ctx context.Context, logger log.Logger, cfg *config.Config
...
+			} else {
+				return err
```
Comment (eseliger) [inline enterprise/cmd/executor/internal/run/run.go:73]: potentially wrap for better user context

File: enterprise/cmd/executor/internal/run/install.go [hunk 3]
```diff
@@ -290,14 +291,24 @@ func installCNIPlugins(cliCtx *cli.Context) error {
...
 
-	// TODO: Not only use srccli.MinimumVersion, but actually the maximum suitable version, fetched from the backend.
-	return download.ArchivedExecutable(cliCtx.Context, fmt.Sprintf("https://github.com/sourcegraph/src-cli/releases/download/%s/src-cli_%s_%s_%s.tar.gz", srccli.MinimumVersion, srccli.MinimumVersion, runtime.GOOS, runtime.GOARCH), path.Join(binDir, "src"), "src")
+	telemetryOptions := newQueueTelemetryOptions(cliCtx.Context, config.UseFirecracker, logger)
+	copts := queueOptions(config, telemetryOptions)
+	client, err := apiclient.NewBaseClient(copts.BaseClientOptions)
+	if err != nil {
+		return err
+	}
+	srcVersion, err := latestSrcCLIVersion(cliCtx.Context, client, copts.BaseClientOptions.EndpointOptions)
```
Comment (eseliger) [inline enterprise/cmd/executor/internal/run/install.go:306]: wow we already had a good part of this?? 🤦 😆

**Review by eseliger — APPROVED** [review]
good tests :)

File: enterprise/cmd/executor/internal/run/install.go (hunk 3 above)
Comment (Piszmog) [reply enterprise/cmd/executor/internal/run/install.go:306]: Yea, I went into thinking I would need to build this. So I decided to add tests to the validate and log a warning on the patch mismatch. Then I was like "ahh, I can use this"

**Review by davejrt — APPROVED** [review]

**Pull request merged**